src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorTableValueWithServiceProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public class ConnectorTableValueWithServiceProvider : ConnectorTableValue
+    {
+        internal IServiceProvider ServiceProvider { get; }
+
+        public ConnectorTableValueWithServiceProvider(ConnectorTableValue tableValue, IServiceProvider serviceProvider)
+            : base(tableValue.Name, tableValue._tabularFunctions, (tableValue.IRContext.ResultType as ConnectorTableType).RecordType)
+        {
+            ServiceProvider = serviceProvider;
+        }
+
+        private IEnumerable<DValue<RecordValue>> _cachedRows;
+
+        protected override async Task<IEnumerable<DValue<RecordValue>>> GetRowsInternal()
+        {
+            if (_cachedRows != null)
+            {
+                return _cachedRows;
+            }
+
+            BaseRuntimeConnectorContext connectorContext = ServiceProvider.GetService<BaseRuntimeConnectorContext>();
+            FormulaValue rows = await _getItems.InvokeAsync(Array.Empty<FormulaValue>(), connectorContext, CancellationToken.None).ConfigureAwait(false);
```

Comment:
>await _getItems.InvokeAsync(Array.Empty<FormulaValue>(), connectorContext, CancellationToken.None).ConfigureAwait(false); [](http://example.com/codeflow?start=32&length=121)

Notice that there is no paging here
Just get 1 page only

Reply:
that's great. 
We probably want to give delegation warnings , similar to what we do on dataveres side.